src/components/RichTextEditor.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { EditorContent, useEditor } from '@tiptap/react';
+import DOMPurify from 'dompurify';
+
+import StarterKit from '@tiptap/starter-kit';
+import TaskItem from '@tiptap/extension-task-item';
+import TaskList from '@tiptap/extension-task-list';
+import Highlight from '@tiptap/extension-highlight';
+import TextAlign from '@tiptap/extension-text-align';
+
+// eslint-disable-next-line import/extensions
+import MenuBar from './richTextEditor/MenuBar';
+
+type Props = {
+  handleSubmit: (content: string) => void;
+};
+
+const RichTextEditor: React.FC<Props> = function ({ handleSubmit }) {
+  const editor = useEditor({
+    extensions: [
+      StarterKit.configure({}),
+      TextAlign.configure({
+        types: ['heading', 'paragraph'],
+      }),
+      TaskList,
+      TaskItem,
+      Highlight,
+    ],
+    content: ``,
```

Comment:
content should come as a prop, and if you do that I believe that

and according to this 
![image](https://github.com/MohammedAlabd/social-network-nextjs/assets/55356198/2b086429-aabc-4441-91f8-e2bb1e0be3e9)

it could be empty, so you will have to make it optional